libs/platform/src/lib/list/list.component.ts

```diff
@@ -375,76 +405,18 @@ export class ListComponent extends CollectionBaseInput implements OnInit, AfterV
         });
     }
 
-    /** @hidden */
-    /** Keyboard manager on list items, set values when passed via array */
+    /**
+     * @hidden
+     * Keyboard manager on list items, set values when passed via array
+     */
     ngAfterViewInit(): void {
         this._keyManager = new FocusKeyManager<BaseListItem>(this.listItems).withWrap();
 
-        if (this.listItems.length !== 0) {
-            this.listItems.first.listItem.nativeElement.setAttribute('tabindex', 0);
-        }
-        this.listItems.forEach((item) => {
-            if (item.navigationIndicator || item.listType === 'detail') {
-                this._partialNavigation = true;
-            }
-        });
-        this.listItems.forEach((item) => {
-            if (!this._partialNavigation) {
-                item.navigated = this.navigated;
-                item.navigationIndicator = this.navigationIndicator;
-                item.listType = this.listType;
-            }
-            item.contentDensity = this.contentDensity;
-            item._isCompact = this._isCompact;
-            item.selectionMode = this.selectionMode;
-            item.selectRow = this.selectRow;
-            item._hasByLine = this.hasByLine;
-            this.stateChanges.next(item);
-        });
-
-        const indicators = this.itemEl.nativeElement.querySelectorAll('fd-busy-indicator');
-        indicators.forEach((indicator) => {
-            if (indicator) {
-                indicator.setAttribute('aria-label', '');
-            }
-        });
-    }
-
-    /** @hidden */
-    /** Setting values from list to list items
-     * example:
-     * Does list item has navigation,
-     * should show arrows,
-     * will it be compact mode,
-     * should be in which selection mode
-     * set values when passed via datasource
-     */
-    ngAfterContentInit(): void {
-        this._itemsSubscription = this.listItems.changes.subscribe((items) => {
-            if (this.listItems.length !== 0) {
-                this.listItems.first.listItem.nativeElement.setAttribute('tabindex', 0);
-            }
+        this._updateListItems();
+        this.listItems.changes.subscribe(() => this._updateListItems());
```

Comment:
takeUntil

Reply:
As discussed, this is not an issue, since it's subscription will be closed automatically